ramips: mt7621: add Cudy X6 v2

Djfe opened the pull request:
This PR adds support for the Cudy X6 v2 and at the same time improves support for v1:
- Corrects the WiFi MAC addresses to match oem/stock behavior.
- Also makes use of the red led: default for failsafe and upgrade (instead of the blue led)

This adds a common .dtsi file for both devices. The .dts files themselves only differ due to the smaller flash size of v2 which causes the partitions to be at different offsets.
______________

About the WiFi Mac correction:
v1 and v2 assign the MAC for 5GHz WiFi slightly differently.
Both set the LA bit, but v1 then adds 0x100000 while v2 adds 0x300000. (at least this is the case for the two devices I own)
OpenWrt now adds 0x100000 in all cases, which is fine I think (the LA bit is set after all).
The previously used mac addresses from the factory partition were completely wrong and belong to Ralink instead of Cudy.

EDIT:
`ramips: Cudy X6 fixes / improvements`
was already merged in 45cf200b2e22c34f2ae043b87e24230de526fefc

The only remaining part is merging the support for X6 v2

blocktrron commented:
Please add a short overview which states the full MAC-address in OpenWrt and stock-firmware respectively.

Djfe commented:
Sure, I updated the commit message and here you go for the full comparison:

stock firmware X6 v1:
```
use   address             source
LAN   b4:4b:d6:2d:c8:4a   label
WAN   b4:4b:d6:2d:c8:4b   label + 1
2g    b4:4b:d6:2d:c8:4a   label
5g    b6:4b:d6:3d:c8:4a   label + LA-Bit set + 4th oktet increased
```
openwrt 5GHz mac:
```
5g    b6:4b:d6:3d:c8:4a
```
______________
stock firmware X6 v2:
```
use   address             source
LAN   f4:a4:54:86:75:a2   label
WAN   f4:a4:54:86:75:a3   label + 1
2g    f4:a4:54:86:75:a2   label
5g    f6:a4:54:b6:75:a2   label + LA-Bit set + 4th oktet increased
```
openwrt 5GHz mac:
```
5g    f6:a4:54:96:75:a2
```

blocktrron commented on target/linux/ramips/dts/mt7621_cudy_x6-v1.dts:
```diff
@@ -0,0 +1,62 @@
+	model = "CUDY X6 v1";
+
+	aliases {
+		label-mac-device = &gmac0;
```
Can be put in dtsi

blocktrron commented on target/linux/ramips/dts/mt7621_cudy_x6.dtsi:
```diff
@@ -6,15 +6,11 @@
 	aliases {
 		led-boot = &led_internet_blue;
-		led-failsafe = &led_internet_blue;
+		led-failsafe = &led_internet_red;
```
Put this change into separate commit, presumably befoe adding v2

blocktrron commented on target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:
```diff
@@ -27,9 +27,11 @@ case "$board" in
+	cudy,x6-v2)
+		hw_mac_addr="$(get_mac_label)"
+		[ "$PHYNBR" = "1" ] && \
+		macaddr_setbit_la "$(macaddr_add $hw_mac_addr 0x100000)" > /sys${DEVPATH}/macaddress
```
Fix the MAC address for v1 before introducing v2.

Djfe committed:
ramips: mt7621: add support for Cudy X6 v2

Rename existing device to v1 and create common .dtsi
Difference to v1: 16MB Flash

Specifications:

SoC: MediaTek MT7621
RAM: 256 MB
Flash: 16 MB (SPI NOR, XM25QH128C on my device)
WiFi: MediaTek MT7915E
Switch: 1 WAN, 4 LAN (Gigabit)
Buttons: Reset, WPS
LEDs: Two Power LEDs (blue and red; together they form purple)
Power: DC 12V 1A center positive
Serial: 115200 8N1
        C440 - (3V3 - GND - RX - TX) - C41 | v1 and v2
               (P   - G   - R  - T)        | v2 labels them on the board
Installation:

Download and flash the manufacturer's built OpenWrt image available at
http://www.cudytech.com/openwrt_software_download
Install the new OpenWrt image via luci (System -> Backup/Flash firmware)
Be sure to NOT keep settings.

Recovery:

Loads only signed manufacture firmware due to bootloader RSA verification
Serve tftp-recovery image as /recovery.bin on 192.168.1.88/24
Connect to any lan ethernet port
Power on the device while holding the reset button
Wait at least 8 seconds before releasing reset button for image to
download

MAC addresses as verified by OEM firmware:

use   address             source
LAN   f4:a4:54:86:75:a2   label
WAN   f4:a4:54:86:75:a3   label + 1
2g    f4:a4:54:86:75:a2   label
5g    f6:a4:54:b6:75:a2   label + LA-Bit set + 4th oktet increased

The label MAC address is found in bdinfo 0xde00.

Signed-off-by: Felix Baumann <felix.bau@gmx.de>

blocktrron commented:
Applied the fixes for rev A to master.

blocktrron commented:
Merged, thanks!
